resources/lib/UnityGroup.php

```diff
@@ -361,9 +362,13 @@ public function getGroupMemberUIDs()
 
     public function requestExists($user)
     {
-        foreach ($this->getRequests() as $requester) {
-            if ($requester->getUID() == $user->getUID()) {
-                return true;
+        $requesters = $this->getRequests();
+        if (count($requesters) > 0) {
+            $requesters = $requesters[0];
+            foreach ($requesters as $requester) {
+                if ($requester->getUID() == $user->getUID()) {
+                    return true;
+                }
```

Comment:
I'm a little confused how this part works. Specifically, the line
```
$requesters = $requesters[0];
```

since `getRequests()` will return an array of tuples, getting the first index of that array will just get you the first tuple, but in this case you want the first element of each tuple. Something like this:

```suggestion
        $requesters = $this->getRequests();
        if (count($requesters) > 0) {
            foreach ($requesters as $requester) {
                if ($requester[0]->getUID() == $user->getUID()) {
                    return true;
                }
```

Reply:
Yeah, you're right, I made a mistake. It should be the way you have suggested not the way I had done. I will commit the  suggested change.